Sources/MapboxDirections/VisualInstructionComponent.swift

```diff
@@ -171,6 +172,64 @@ public extension VisualInstruction.Component {
     }
 }
 
+/**
+ A mapbox shield representation of a visual instruction component.
+ */
+public struct MapboxShield: Equatable, Codable {
+    /**
+     Initializes a mapbox shield with the given name, text color, and display ref.
+     */
+    public init(baseURL: URL, name: String, textColor: String, displayRef: String) {
+        self.baseURL = baseURL
+        self.name = name
+        self.textColor = textColor
+        self.displayRef = displayRef
+    }
+    
+    /**
+     Base URL to query the styles endpoint.
+     */
+    public let baseURL: URL
+
+    /**
+     String indicating the name of the route shield.
+     */
+    public let name: String
+    
+    /**
+     String indicating the color of the text to be rendered on the route shield.
+     */
+    public let textColor: String
```

Comment:
Yes let's resolve it here. The current shield text color is limited to 7 colors (black, blue, green, orange, red, white, and yellow), and these are not standard CSS colors.

Reply:
Oh, my bad, I was confusing this property with the property in the shield spritesheet metadata file, which can contain CSS color values. The keywords you’re referring to are meant to be resolved based on the map or UI style. Since it can differ by context, a concrete UIColor/NSColor would be too inflexible here.

If the set of possible values is a closed set that we don’t foresee being extended in the future, we could make it into an enumeration. (The only standard sign color I don’t see represented here is purple, but I don’t think it ever occurs as a foreground text color, only a background text color.